build_runner/README.md

```diff
@@ -48,99 +47,15 @@ dev_dependencies:
+Builder to decide whether the build needs to be customized. If it does you may
+also provide a `build.yaml` with the configuration. See the
+`package:build_config` README for more information on this file.
 
-Your script should use one of the two functions defined by this library:
-
-- [**`build`**][build_fn]: Run a single build and exit.
-- [**`watch`**][watch_fn]: Continuously run builds as you edit files.
-
-### Configuring
-
-Both [`build`][build_fn] and [`watch`][watch_fn] have a single required
-argument, a `List<BuildAction>` - each of these [`BuildAction`][build_action]s
-may run in parallel, but they may only read outputs from steps _earlier_ in the
-list.
-
-A [`BuildAction`][build_action] is a combination of a single
-[`Builder`][builder], which actually generates outputs, and a single
-[`InputSet`][input_set] , which determines what the primary inputs to that
-builder will be.
-
-Lets look at a very simple example, with a single `BuildAction`. You can ignore
-the `CopyBuilder` for now, just know that its a `Builder` which copies files:
-
-```dart
-main() async {
-  await build([
-    new BuildAction(
-      new CopyBuilder('.copy'), 
-      'my_package', 
-      inputs: ['lib/*.dart'],
-    ),
-  ]);
-}
-```
-
-> Copies all `*.dart` files under `lib` to a corresponding `*.dart.copy`.
-
-Every time you run a `build`, `build_runner` checks for any changes to the
-inputs specified, and reruns the `CopyBuilder` for the inputs that actually
-changed.
-
-A build with multiple steps may look like:
-
-```dart
-main() async {
-  await build([
-    new BuildAction(
-      new CopyBuilder('.copy1'), 
-      'my_package', 
-      inputs: ['lib/*.dart'],
-    ),
-    new BuildAction(
-      new CopyBuilder('.copy2'), 
-      'my_package', 
-      inputs: ['lib/*.dart'],
-    ),
-  ]);
-}
-```
-
-> Makes _two_ copies of every `*.dart` file, a `.copy1` and `.copy2`.
-
-Let's say, however, you want to use the _previous_ output as an input to the
-next build action - for example, you want to convert `.md` (Markdown) to
-`.html`, and then convert `.html` into a `.png` (screenshot of the page):
-
-```dart
-main() async {
-  await build([
-    new BuildAction(
-      new MarkdownToHtmlBuilder(), 
-      'my_package', 
-      inputs: ['web/**.md'],
-    ),
-    new BuildAction(
-      new ChromeScreenshotBuilder(), 
-      'my_package', 
-      inputs: ['web/**.html'],
-    ),
-  ]);
-}
-```
-
-> This time, all the `*.html` files will be created first, and since the next
-> `BuildAction` is waiting for `*.html` inputs, it will _wait_ for them to be
-> created, and then create the `*.png` (screenshot) files.
-
-**NOTE**: Any time you change your build script (or any of its dependencies),
-the next build will be a full rebuild. This is because the system has no way
-of knowing how that change may have affected the outputs.
+Run `pub run build_runner serve` to start up a development server compiling with
```

Comment:
we should probably mention the other commands here as well?

Reply:
Synced offline about this. These commands are in a strange state depending on who the target user is. We have a lot of open questions to answer and we'll wait to expand the docs until we have more answers.